opal/class/opal_lifo.h

```diff
@@ -142,10 +167,8 @@ static inline opal_list_item_t *opal_lifo_pop_atomic (opal_lifo_t* lifo)
     opal_list_item_t *item;
 
     do {
-
-        old_head.data.counter = lifo->opal_lifo_head.data.counter;
-        opal_atomic_rmb ();
-        old_head.data.item = item = (opal_list_item_t*)lifo->opal_lifo_head.data.item;
+        opal_read_counted_pointer (&lifo->opal_lifo_head, &old_head);
```

Comment:
Do you need an opal_atomic_rmb() here after the opal_read_counted_pointer(), or is there a read memory barrier even if that function gets inlined?

Reply:
Should be ok since there is a rmb in opal_read_counted_pointer and old_head.data.item is a data dependency for the next line. Shouldn't be re-ordered by the compiler.

Reply:
thanks for taking a second look.
looks good to me.